modules/ROOT/pages/common-commands-troubleshoot.adoc

```diff
@@ -0,0 +1,96 @@
+
+* <<retrieve-adapter-logs,Retrieve adapter logs>>
+* <<access-the-adapter,Access the adapter>>
+* <<retrieve-policies-and-contracts-info,Retrieve policies and contract information from the adapter>>
+* <<modify-log4j-config,Modify log4j configuration>>
+* <<replace-service-mesh-license,Replace Anypoint Service Mesh license>>
+
+[[retrieve-adapter-logs]]
+== Retrieve Adapter Logs
+
+To retrieve the adapter logs when you are troubleshooting an issue:
+
+. Display the adapter pod:
++
+`$ kubectl get pods -n service-mesh`.
++
+An output is displayed that looks like this:
++
+//add text of the screenshot here
+. Retrieve the adapter logs:
+`$ kubectl -n service-mesh logs $(kubectl -n service-mesh get pod -l namespace=<replace_with_namespace> -oname) -c mule`.
+
+[[access-the-adapter]]
+== Access the Adapter
+
+When troubleshooting Anypoint Service Mesh, you might need to access and display the information about the adapter.
+
+//[PA: Kim is this format better or the ones that I follow in the next few topics?]
+To display information about the adapter pods:
+
+. Dispalys all the pods created under the namespace service-mesh:
++
+`$ kubectl get pods -n service-mesh`.
++
+An output is displayed that looks like this:
++
+//add text for the screenshot here
+. Access the adapter:
++
+`$ kubectl -n service-mesh exec -ti  <adapter-pod-name>`.
+
+[[retrieve-policies-and-contracts-info]]
+== Retrieve Policies and Contracts Information from the Adapter
+
+When troubleshooting issues with Anypoint Service Mesh, you might need to retrieve policies and contracts details:
+
+. Display the adapter pod: 
++
+`$ kubectl get pods -n service-mesh`.
++
+An output is displayed that looks like this:
++
+//add text for the screenshot here
+. Download the policy and contract files:
++
+`$ kubectl cp service-mesh/$(kubectl -n service-mesh get pod -ocustom-columns='NAME:.metadata.name' --no-headers -l namespace=<replace_with_namespace>):/usr/src/app/policies -c mule ~/Desktop/`.
+
+[[modify-log4j-config]]
+== Modify Log4j configuration
+
+When troubleshooting Anypoint Service Mesh, you might need to modify the Log4j configuration:
+
+. Execute the following command to display the adapter pod:
++
+`$ kubectl get pods -n service-mesh`.
++
+An output is displayed that looks like this:
++
+//add text for the screenshot there
+. Download the current log4j configuration:
++
+`$ kubectl cp -c mule service-mesh/<replace_with_grpcmule_pod_name>:/usr/src/app/conf/log4j2.xml log4j2.xml`.
+. Update the downloaded copy of log4j2.xml as required and upload it again to the pod:
++
+`$ kubectl cp -c mule log4j2.xml service-mesh/<replace_with_grpcmule_pod_name>:/usr/src/app/conf/log4j2.xmlonf/log4j2.xml`.
++
+The `log4j` configuration file is configured to check for changes every 60 seconds. Therefore to view the modifications applied, you might have to wait for that amount of time at a minimum.
+
+[[replace-service-mesh-license]]
+== Replace the Service Mesh license
```

Comment:
```suggestion
== Replace the Service Mesh License
```

Reply:
Should this really be Update the Service Mesh License?
The first step implies that the existing license is not valid, but there might be other reasons that the license is not appropriate?

Reply:
Mostly not. But changed it to "Update". Done! Thanks!